third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -114,61 +123,387 @@ lazy_static! {
+            Self::V2(loader) => loader
+                .check_dependencies_and_charge_gas(
+                    module_storage,
+                    gas_meter,
+                    visited,
+                    referenced_modules,
+                    ids,
+                )
+                .map_err(|e| e.finish(Location::Undefined)),
+        }
+    }
+
+    pub(crate) fn load_script(
+        &self,
+        script_blob: &[u8],
+        ty_args: &[TypeTag],
+        data_store: &mut TransactionDataCache,
+        module_store: &ModuleStorageAdapter,
+        module_storage: &impl ModuleStorageV2,
+        script_storage: &impl ScriptStorage,
+    ) -> VMResult<(Arc<Script>, LoadedFunction)> {
+        match self {
+            Self::V1(loader) => loader.load_script(script_blob, ty_args, data_store, module_store),
+            Self::V2(loader) => loader
+                .load_script(module_storage, script_storage, script_blob, ty_args)
+                .map_err(|e| e.finish(Location::Undefined)),
+        }
+    }
+
+    pub(crate) fn load_module(
+        &self,
+        id: &ModuleId,
+        data_store: &mut TransactionDataCache,
+        module_store: &ModuleStorageAdapter,
+        module_storage: &impl ModuleStorageV2,
+    ) -> VMResult<Arc<Module>> {
+        match self {
+            Loader::V1(loader) => loader.load_module(id, data_store, module_store),
+            Loader::V2(loader) => loader
+                .load_module(module_storage, id.address(), id.name())
+                .map_err(|e| e.finish(Location::Undefined)),
+        }
+    }
+
+    fn load_function_without_type_args(
+        &self,
+        module_id: &ModuleId,
+        function_name: &IdentStr,
+        data_store: &mut TransactionDataCache,
+        module_store: &ModuleStorageAdapter,
+        module_storage: &impl ModuleStorageV2,
+    ) -> VMResult<Arc<Function>> {
+        match self {
+            Loader::V1(loader) => {
+                // Need to load the module first, before resolving the function.
+                loader.load_module(module_id, data_store, module_store)?;
+                module_store
+                    .resolve_function_by_name(function_name, module_id)
+                    .map_err(|err| err.finish(Location::Undefined))
+            },
+            Loader::V2(loader) => loader
+                .load_function_without_ty_args(
+                    module_storage,
+                    module_id.address(),
+                    module_id.name(),
+                    function_name,
+                )
+                .map_err(|e| e.finish(Location::Undefined)),
+        }
+    }
+
+    // Loading verifies the module if it was never loaded.
+    // Type parameters are inferred from the expected return type. Returns an error if it's not
+    // possible to infer the type parameters or return type cannot be matched.
+    // The type parameters are verified with capabilities.
+    pub(crate) fn load_function_with_type_arg_inference(
```

Comment:
is this unchanged? it bloats the PR and might be harder to review (since it's not side to side), but also does make sense logically to move in the file. If there is a lot of it, might make sense to keep them at original place so github won't mark as new code?

Reply:
This is unchanged yes, I only changed the internal function that loads without type arguments. What I can do is just add impl Loader at the right places and then GitHub should detect it is the same code. Let me try that.

Reply:
Fixed! Now the diff looks better. I still kept `match_return_type` function in the end - that one is just a copy paste and it doesn't have to be a class member.